telemetry/csharp/ToolkitTelemetryGenerator/Models/MetricType.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace ToolkitTelemetryGenerator.Models
+{
+    public class MetricType
+    {
+        public string name { get; set; }
+        public string type { get; set; }
+        public string description { get; set; }
+        public string[] allowedValues { get; set; }
```

Comment:
For modeling sake it probably makes sense to make allowedValues nullable since they are not required and it does a different thing when they are not there: when it is empty I don't think it captures that as well.

Reply:
Strings and arrays are inherrently nullable in C#. I missed a null check, which I'll add.